### PR TITLE
Deprecate Data.Array.Foreign.* to Streamly.Internal.Data.Array.Unboxed.*

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
this PR is replacement of : https://github.com/composewell/streamly/pull/1694

File: core/streamly-core.cabal [hunk 1]
```diff
@@ -278,9 +278,10 @@ library
                     -- Unboxed IORef
                      , Streamly.Internal.Data.IORef.Unboxed
                      -- May depend on streamly-core-stream
-                     , Streamly.Internal.Data.Array.Foreign.Mut.Type
-                     , Streamly.Internal.Data.Array.Foreign.Mut
-                     , Streamly.Internal.Data.Array.Foreign.Type
+                     , Streamly.Internal.Data.Array.Unboxed
```
Comment (harendra-kumar) [inline core/streamly-core.cabal:281]: Move this after line 327.

File: streamly.cabal [hunk 2]
```diff
@@ -202,7 +202,7 @@ extra-source-files:
     core/src/Streamly/Internal/Data/Stream/Zip.hs
     core/src/Streamly/Internal/Data/List.hs
     core/src/Streamly/Internal/Data/Array.hs
-    core/src/Streamly/Internal/Data/Array/Foreign.hs
+    core/src/Streamly/Internal/Data/Array/*.hs
```
Comment (harendra-kumar) [inline streamly.cabal:205]: Better not use `*` as it might pick up any stale, backup files lying around in the workspace.

File: streamly.cabal [hunk 3]
```diff
@@ -615,7 +616,7 @@ library
                      -- streamly-core-data-arrays
                      -- XXX Depends on primitive
                      , Streamly.Internal.Data.Array
-                     , Streamly.Internal.Data.Array.Foreign
+                     , Streamly.Internal.Data.Array.Unboxed
```
Comment (harendra-kumar) [inline streamly.cabal:619]: We have removed the Foreign module completely. Need to only deprecate it, not remove.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Pull request merged**